translation-service-eTranslation/src/main/java/eu/europeana/api/translation/service/etranslation/ETranslationTranslationService.java

```diff
@@ -0,0 +1,211 @@
+    //base64 encoded string, which will not be translated from eTransaltion, used as a markup delimiter
+    String markupDelimit=" ZXd3ZHdld2U= ";
+    String eTranslJointStr = generateJointStringForTranslation(translationObjs, markupDelimit);
+    
+    /* create an eTransl request with an external reference and send it. The same external reference is received
+     * in the eTransl callback. That reference is used for the name of the channel for the redis message subscriber 
+     * listener created below, which will be notified from the redis publisher after the eTransl callback comes.
+     * The publisher will publish to the same channel using the external reference from the eTransl callback.
+     */
+    if(StringUtils.isNotBlank(eTranslJointStr)) {
+      //create external reference for eTransl service
+      byte[] eTranslExtRefBase64 = null;
+      try {
+        eTranslExtRefBase64 = Base64.encodeBase64(eTranslJointStr.getBytes(StandardCharsets.UTF_8.name()));
+      } catch (UnsupportedEncodingException e) {
+        throw new TranslationException("Exception during the eTranslation base64 encoding for the external reference.", 0, e);
+      }
+      String eTranslExtRef = new String(eTranslExtRefBase64, StandardCharsets.UTF_8);
+
+      //create and send the eTransl request
+      //baseUrl is empty for the integration tests, where the eTranslation service will not be called
+      if(!StringUtils.isBlank(baseUrl)) {
+        try {
+          String body = createTranslationBody(eTranslJointStr,translationObjs.get(0).getSourceLang(),translationObjs.get(0).getTargetLang(),eTranslExtRef);
+          createHttpRequest(body);
+        } catch (JSONException e) {
+          throw new TranslationException("Exception during the eTranslation http request body creation.", 0, e);
+        } catch (IOException e) {
+          throw new TranslationException("Exception during sending the eTranslation http request.", 0, e);
+        }  
+      }    
+      
+      //create a redis message listener obj, and wait on that obj until it get notified from the redis publisher
+      createRedisMessageListenerAndWaitForResults(translationObjs, eTranslExtRef, markupDelimit);
+      
+    }
+  }
+  
+  private void createRedisMessageListenerAndWaitForResults(List<TranslationObj> translationObjs, String eTranslExtRef, String markupDelimit) throws TranslationException {
+    RedisMessageListener redisMessageListener = new RedisMessageListener();
+    MessageListenerAdapter redisMessageListenerAdapter = new MessageListenerAdapter(redisMessageListener);
+    redisMessageListenerContainer.addMessageListener(redisMessageListenerAdapter, ChannelTopic.of(eTranslExtRef));
+    synchronized (redisMessageListener) {
+      //while loop as a good practice to ensure spurious wake-ups (https://www.baeldung.com/java-wait-notify)
+      while(redisMessageListener.getMessage()==null) {
+        try {
+          redisMessageListener.wait(maxWaitMillisec);
```

Comment:
there is a flow here, as the false notifications will reset the overall timeout ... we might need to address this by using timestamps